enclave_wrangler/datasets.py

```diff
@@ -0,0 +1,355 @@
+        concept_csm = df.merge(csm_df, on='concept_id')
+        df = df[df['concept_id'].isin(concept_csm['concept_id'])]
+    except FileNotFoundError:
+        print('Warning: Tried transforming concept.csv, but concept_set_members.csv must be downloaded and transformed '
+              'first. Try running again after this process completes. Eventually need to do these transformations '
+              'dependency ordered fashion.', file=sys.stderr)
+
+    return df
+
+
+def transform_dataset__concept_ancestor(dataset_name: str) -> pd.DataFrame:
+    """Transformations to concept_ancestor.csv"""
+    df = pd.read_csv(os.path.join(CSV_DOWNLOAD_DIR, dataset_name + '.csv'), keep_default_na=False).fillna('')
+    df = transforms_common(df)
+
+    # JOIN
+    try:
+        csm_df = pd.read_csv(
+            os.path.join(CSV_TRANSFORM_DIR, 'concept_set_members.csv'), keep_default_na=False).fillna('')
+        ca_csm = df.merge(csm_df, left_on='ancestor_concept_id', right_on='concept_id')
+        df = df[df['ancestor_concept_id'].isin(ca_csm['ancestor_concept_id'])]
+    except FileNotFoundError:
+        print('Warning: Tried transforming concept_ancestor.csv, but concept_set_members.csv must be downloaded and '
+              'transformed first. Try running again after this process completes. Eventually need to do these '
+              'transformations dependency ordered fashion.', file=sys.stderr)
+
+    return df
+
+
+def transform_dataset__concept_set_members(dataset_name: str) -> pd.DataFrame:
+    """Transformations to concept.csv"""
+    df = pd.read_csv(
+        os.path.join(CSV_DOWNLOAD_DIR, dataset_name + '.csv'),
+        # dtype={'archived': bool},    # doesn't work because of missing values
+        converters={'archived': lambda v: v and True or False},  # this makes it a bool field
+        keep_default_na=False).fillna('')
+    df = transforms_common(df)
+
+    return df
+
+
+def transform_dataset__code_sets(dataset_name: str) -> pd.DataFrame:
+    """Transformations to code_sets.csv"""
+    df = pd.read_csv(os.path.join(CSV_DOWNLOAD_DIR, dataset_name + '.csv'), keep_default_na=False).fillna('')
+    df = transforms_common(df)
+
+    # JOIN
+    try:
+        csc_df = pd.read_csv(
+            os.path.join(CSV_TRANSFORM_DIR, 'concept_set_container_edited.csv'), keep_default_na=False).fillna('')
+        container_concept_set_name_ids = set(csc_df['concept_set_id'])
+        df = df[df['concept_set_name'].isin(container_concept_set_name_ids)]
+    except FileNotFoundError:
+        print('Warning: Tried transforming code_sets.csv, but concept_set_container_edited.csv must be downloaded and '
+              'transformed first. Try running again after this process completes. Eventually need to do these '
+              'transformations dependency ordered fashion.', file=sys.stderr)
+
+    return df
+
+
+def transforms_common(df: pd.DataFrame) -> pd.DataFrame:
```

Comment:
A function for common transformations. Ideally I'd read the file in here as well to avoid duplication, but one of the transformations (concept set members) involves some special operations on file read, so for now I am doing a file read in every function.